csrc/host_ir/jit.cpp

```diff
@@ -1028,13 +1147,22 @@ void HostIrJitImpl::registerExternalFunctions() {
       +[]() -> at::Tensor* { return new at::Tensor(); });
 
   // in place tensor update
-  void* set_tensor_func_ptr =
+  void* set_tensor_out_func_ptr =
       reinterpret_cast<void*>(+[](at::Tensor* out, at::Tensor* in) -> void {
         NVF_ERROR(out != nullptr, kSetTensorFuncName, " out is nullptr");
         NVF_ERROR(in != nullptr, kSetTensorFuncName, " in is nullptr");
         *out = in->clone(); // Clone the input tensor
       });
 
+     // copy and return tensor
+   void* set_tensor_func_ptr =
+       reinterpret_cast<void*>(+[](at::Tensor* in) -> at::Tensor* {
+         NVF_ERROR(in != nullptr, kSetTensorFuncName, " in is nullptr");
+         at::Tensor* out = new at::Tensor();
+         *out = in->clone();
```

Comment:
```suggestion
         *out = *in;
```
Do you need a clone? That's more memory and time.

Reply:
done, for already allocated out, I followed host ir evaluator protocol to make a `copy_` so that we don't just simply throw away original buffer.